docs/dev/developer-guides/transactions/blocks.md

```diff
@@ -0,0 +1,68 @@
+
+## Prerequisite
+
+To fully understand this page, we suggest reading [accounts](account.md), [transactions] and (transactions.md).
+
+## How do Ethereum blocks work?
+
+Blocks are strictly ordered (each new block has a connection to its parent block), and transactions within blocks are strictly ordered to keep track of the history of transactions.
+Unless there are exceptional circumstances, all network participants agree with the precise number and history of blocks at any particular time and attempt to combine the pending live transaction requests into the upcoming block.
+
+
+Every node adds the newly created block to the end of its blockchain after it has been assembled by one of the network's validators, and a new validator is chosen to build the following block.
+
+
+## Block numbers and time
+
+On zkSync, when a user sends L2 transfers to the operator, they are not processed immediately.
+Instead, they are included in a group of pending transactions in the available off-chain state. The user receives witnesses to their balance and all pending transactions in the new rollup state when the operator updates the rollup state. The blockchain does not contain these witnesses.
+The user signs and sends an operation labeled `ProcessTransactions` to the operator to have their pending transactions processed.
+The operator then includes this operation into the subsequent rollup block, which completes all pending transactions for the sender and changes the on-chain available state to `lastSeenBlockNum(sender) = blockNum`, where `blockNum` is the number of the most recent block. The operator provides witnesses for the updated balances of all users impacted by a rollup block once it has been posted.
+
+
+### Block Numbers: Ethereum vs. zkSync
+
+The structure of L2 blocks is similar to that of L1 blocks. 
+In zkSync, we send to the L1 contract some information about L2 blocks (a commitment), then we produce a proof for the blocks and transmit it back to the L1 contract, which confirms that the proof is accurate.
+
+
+### Ethereum Block Numbers Within zkSync
+
+Accessing block numbers within a zkSync smart contract (i.e., `block. number` in Solidity) is similar to how you would do it on Ethereum. For example, `eth_blockNumber` returns the number of the latest L2 block, and `eth_getBlockByNumber` returns the information about the L2 block by its number.
+
+The **capacity,** which includes the number of transactions and storage writes, of the L1 batch is limited because of two different things:
+
+
+1. For small L1 batches, a prover can produce proofs in an adequate amount of time.
+2. The required gas for the transaction to commit the L1 batch on the L1 contract must not exceed the permitted gas for the L1 block. Because that gas depends on L1 batch size, capacity is constrained.
+
+
+L1 batches are sealed for two reasons:
+
+1. The capacity is reached (well, because we can't send an L1 batch that exceeds capacity).
```

Comment:
@perekopskiy what is the capacity that can be reached?

Reply:
You can add information that Stat suggests to the paragraph above:
I mean instead of 

1. For small L1 batches, a prover can produce proofs in an adequate amount of time.
2. The required gas for the transaction to commit the L1 batch on the L1 contract must not exceed the permitted gas for the L1 block. Because that gas depends on L1 batch size, capacity is constrained.

write

1. Capacity for zero-knowledge circuits: a prover can produce proofs in an adequate amount of time only for the circuits that fit some conditions.
2. Capacity for the L1 public data: the required gas for the transaction to commit the L1 batch on the L1 contract must not exceed the permitted gas for the L1 block. Because that gas depends on L1 batch size, capacity is constrained.